src/pages/uxp/reference-js/Modules/uxp/Persistent File Storage/FileSystem.md

```diff
@@ -0,0 +1,396 @@
+
+## close(fd, callback)
+Closes a file descriptor asynchronously
+
+| Param | Type | Description |
+| --- | --- | --- |
+| fd | `int` | file descriptor of the file to close |
+| callback | `function` | if not provided, this function will return Promise object |
+
+**Returns**: `int` - 0 if succeeded, otherwise throws an Error  
+
+**Example**  
+```js
+await fs.close(fd);
+```
+
+
+<a name="module-storage-filesystem-read" id="module-storage-filesystem-read"></a>
+
+## read(fd, buffer, offset, length, position, callback)
+Reads data in chunks from the file it refers to the file descriptor
+
+**Throws**:
+- `Error` if invalid file descriptor is passed.
+ if invalid parameter format or value is passed.
+
+
+| Param | Type | Description |
+| --- | --- | --- |
+| fd | `int` | a file descriptor obtained from fs.open |
+| buffer | `ArrayBuffer` | the buffer where read bytes are written to |
+| offset | `int` | the offset to the buffer where read bytes are written from |
+| length | `int` | the length to read |
+| position | `int` | the position of the file to read from. if -1, the current file position to read from. when the bytes are read, the current file position advances by size of the read bytes. if the value is greater than or equal to 0, it specifies a file position to read from. after the bytes are read, a current file position stayed the same |
+| callback | `function` | if not provided, this function will return Promise object |
+
+**Returns**: `Promise<Object>` - { bytesRead, buffer }  
+
+**Example**  
+```js
+const fileSize = 1024;
+const buffer = new ArrayBuffer(fileSize);
+const fd = await fs.open("plugin-data:/fileToRead.txt", "r");
+let bytesReadInTotal = 0;
+while (bytesReadInTotal < fileSize) {
+    const { bytesRead } = await fs.read(fd, buffer, bytesReadInTotal, 128, -1);
+    if (!bytesRead) {
+        break;
+    }
+    bytesReadInTotal += bytesRead;
+}
+await fs.close(fd);
+```
+
+
+<a name="module-storage-filesystem-write" id="module-storage-filesystem-write"></a>
+
+## write(fd, buffer, offset, length, position, callback)
+Writes data in chunks to the file it refers to the file descriptor
+
+**Returns**: `Promise<Object>` - { bytesWritten, buffer }  
+**Throws**:
+
+- `Error` if invalid file descriptor is passed
+if invalid parameter format or value is passed
+
+
+| Param | Type | Description |
+| --- | --- | --- |
+| fd | `int` | the file descriptor obtained from fs.open |
+| buffer | `ArrayBuffer` | the buffer where the data to write with |
+| offset | `int` | the offset of the buffer where write bytes start from |
+| length | `int` | the length to write |
+| position | `int` | the position of the file to write from. if -1,writing will start from the current file position. when the bytes are written, the current file position advances by size of the written bytes. if the value is greater than or equal to 0, it specifies a file position to write from. After writing, it will not change the file position |
```

Comment:
```suggestion
| fd | `number` | the file descriptor obtained from fs.open |
| buffer | `ArrayBuffer` | the buffer where the data to write with |
| offset | `number` | the offset of the buffer where write bytes start from |
| length | `number` | the length to write |
| position | `number` | the position of the file to write from. if -1,writing will start from the current file position. when the bytes are written, the current file position advances by size of the written bytes. if the value is greater than or equal to 0, it specifies a file position to write from. After writing, it will not change the file position |
```

Reply:
Postponed to a "data type normalization" effort